src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -2328,6 +2329,16 @@ private IEnumerable<TypeInferenceInfo> InferTypeInWhileStatement(WhileStatementS
 
                 return CreateResult(SpecialType.System_Boolean);
             }
+
+            private IEnumerable<TypeInferenceInfo> InferTypeInRelationalPattern(RelationalPatternSyntax relationalPattern)
+            {
+                if (relationalPattern.Parent is IsPatternExpressionSyntax isPatternExpression)
+                {
+                    return InferTypeInIsPatternExpression(isPatternExpression, relationalPattern);
+                }
+
+                return SpecializedCollections.EmptyEnumerable<TypeInferenceInfo>();
+            }
```

Comment:
```suggestion
            private IEnumerable<TypeInferenceInfo> InferTypeInRelationalPattern(RelationalPatternSyntax relationalPattern)
                => InferTypes(relationalPattern);
```

I think this will work just as well and fit the general idea of this type (Which is walking up to find the right inference point).

Reply:
Yes, you're right. It's working.